### PR TITLE
adding OS X install script

### PR DESCRIPTION
**Pull request by abhinavmoudgil95**
Added a script for automatic installation on Mac OS X. 
Kindly review.

### CODE REVIEW

File: install-osx.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+sed -i '1i\
+#define TCP_KEEPIDLE 14400\
+' $file
+
+# Edit PyMindAgent.cc to make OS X compatible
+file="./opencog/opencog/cogserver/modules/python/PyMindAgent.cc"
+ssed -i '1i\
+#include <Python.h>\
+' $file
+sed -i '0,/if __GNUC__/s/if __GNUC__/if __GNUC__ \&\& INIT_PRIORITY \&\& ((GCC_VERSION >= 40300) || (CLANG_VERSION >= 20900))/' $file
```
Comment (AmeBel) [inline install-osx.sh]: It would be best if the above two file edits are done in the opencog repo, on the actual c++ code and associated cmake files.

Comment (abhinavmoudgil95) [inline install-osx.sh]: Should I submit a pr in opencog repo by making above changes? We may have to verify that these changes doesn't affect unix installation.

Comment (AmeBel) [inline install-osx.sh]: I am not saying add a script there, rather use cmake macros like https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_APPLE.html to conditionaly define `TCP_KEEPIDLE 14400` and include `Python.h`. 

@inflector do you have better suggestion?

**Comment by AmeBel** [comment]
thanks

**Pull request merged**